[IAP] Update plan selection UI to M2

iamgabrielma opened the pull request:
Closes: #10189 

## Description
With this PR we start the implementation of the updated UI for In-App Purchases plan selection (M2). We mainly move to a card-based layout, with one selectable card for each plan:

Since the UI updates are pretty extensive (and are still ongoing), there are some items we have left out of this PR and we'll work on it separately, like so:
- The `View Full Feature List` button has not been added yet.
- The contents of `View PlanName Features` when the view is expanded have not been added yet.
- The text for "Billed annually/25 per month" while we confirm this won't be a problem, for the moment only shows "per month" or "per year" for each case.
- The button text is currently dynamic and appends Monthly or Yearly depending on which plan is chosen. We're discussing if this needs to be changed here p1688979147117499-slack-C0354HSNUJH and show only the plan name without the frequency.

For clarity, I marked here the part we're working with in this PR:

<img width="597" alt="Screenshot 2023-07-10 at 16 35 13" src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/4191aaf3-df99-4d6a-9e9b-31a1b9eab69f">| 

## Changes
* Updated the UI to a card-based layout, moving from `List` and `Section` to use `ScrollView` and `VStack`
* Added segmented control (monthly, yearly)
* Added a checkbox to reflect selected/unselected plans
* Cards are expandable upon tapping `View PlanName Features` (details to be implemented on a different PR)
* Updated plan description to fit the latest design update
* Added a badge that shows `POPULAR` in Performance plans

## Testing instructions

* On a Free Trial site, eligible for IAP (a new site can be created in Menu > Switch Sites > Add New Site) tap on `Purchase Now` and observe that:
  * No plan is selected when the screen is loaded
  * The primary purchase button should be disabled and read Select a plan
  * When a plan is selected, the primary button changes to "Purchase Essential Monthly" (or the selected plan)
  * Checkbox is visible when a plan is selected
  * The yearly option reads as "Annually (Save 35%)"
  * Performance plans have the "POPULAR" badge
  * Tapping on "View Essential Features" expands and collapses a new section.

## Screenshots

| Screen loaded | Plan selected | Plan details expanded |
|--------|--------|--------|
| ![Simulator Screen Shot - iPhone 11 Pro - 2023-07-10 at 16 58 05](https://github.com/woocommerce/woocommerce-ios/assets/3812076/0a0bd67a-aa1c-4a20-bc1e-77ab8112ecd2) | ![Simulator Screen Shot - iPhone 11 Pro - 2023-07-10 at 16 58 13](https://github.com/woocommerce/woocommerce-ios/assets/3812076/2544bed6-0109-4c74-99d4-c21b6e4e0587) | ![Simulator Screen Shot - iPhone 11 Pro - 2023-07-10 at 16 58 20](https://github.com/woocommerce/woocommerce-ios/assets/3812076/01f88537-56c0-463e-997a-b5f9de89747d) |

joshheald committed:
Work towards M2 designs using a scrolling VStack

We were previously misusing list quite a lot. This moves us to a scrolling VStack of cards.

iamgabrielma committed:
Update to latest copy  for plan descriptions

iamgabrielma committed:
Remove unused string

iamgabrielma committed:
Move strings to NSLocalized

iamgabrielma committed:
Add “popular” plan badge

iamgabrielma committed:
Fix dark mode for card background

iamgabrielma committed:
Update font size and text

iamgabrielma committed:
Update font weight for plan name and price

toupper reviewed (approved):
LGTM :) I tested it mostly successfully, with a few questions:

No plan is selected when the screen is loaded ✅ 
The primary purchase button should be disabled and read Select a plan  ✅ 
When a plan is selected, the primary button changes to "Purchase Essential Monthly" (or the selected plan)  ✅ 
Checkbox is visible when a plan is selected  ✅ 
The yearly option reads as "Pay Annually (Save 35%)" ❌ It reads Annually(Save 35%). It's like that on the PR summary screenshot.
Performance plans have the "POPULAR" badge  ✅ 
Tapping on "View Essential Features" expands and collapses a new section.  ✅ New section just shows a string: Expanded ❓

iamgabrielma commented:
Thanks for the review!

> The yearly option reads as "Pay Annually (Save 35%)" ❌ It reads Annually(Save 35%). It's like that on the PR summary screenshot. 

Yes! I didn't update the PR description, should be `Annually(Save 35%)` as you saw 👍 

> New section just shows a string: Expanded ❓

Yep! I added an `EmptyView()` initially but upon tapping doesn't show anything, so for now just added a temporary `Text` view with that string for it to be tested and see the collapse/expanse visually 👍
